src/core/pointcloud/qgspointcloudlayer.h

```diff
@@ -240,6 +240,27 @@ class CORE_EXPORT QgsPointCloudLayer : public QgsMapLayer, public QgsAbstractPro
      */
     QVariant statisticOf( const QString &attribute, QgsStatisticalSummary::Statistic statistic ) const;
 
+    /**
+     * Returns a list of existing classes which are present for the specified \a attribute.
+     *
+     * \since QGIS 3.26
+     */
+    QVariantList classesOf( const QString &attribute ) const;
+
+    /**
+     * Returns a statistic for one class \a value from the specified \a attribute.
+     *
+     * If no matching precalculated statistic is available then an invalid variant will be returned.
+     * \since QGIS 3,26
+     */
+    QVariant classStatisticOf( const QString &attribute, const QVariant &value, QgsStatisticalSummary::Statistic statistic ) const;
+
+    /**
+     * Returns the object containing all the calculated statistics
+     * \since QGIS 3.26
+     */
+    QgsPointCloudStatistics calculatedStatistics() const { return mCalculatedStatistics; }
```

Comment:
I would suggest to rename this to just `statistics()` - and it would either return stats from the provider or the calculated stats.

Also, please remove the functions statisticOf(), classesOf(), classStatisticOf() - places that need access to stats would simply call `layer->statistics()->someFcn()` - this will keep the point cloud layer API cleaner.